packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -865,59 +903,60 @@ export class GarbageCollector implements IGarbageCollector {
     }
 
     /**
-     * Validates that all new references are correctly identified and processed. The basic principle for validation is
-     * that we should not have new references in the reference graph (GC data) that have not been notified to the
-     * garbage collector via `referenceAdded`.
-     * We validate that the references in the current reference graph should be a subset of the references in the last
-     * run's reference graph + references since the last run.
+     * Finds outbound references to datastores in the current GC graph that aren't in the previous GC graph and returns
```

Comment:
This line is tied to the specific implementation that is done. It should provide a high level view of what this function is doing and then you can mention the current implementation if needed. Or, a better approach is that the implementation part of the comment can move inside the function. Similarly, the part about only doing this for data stores currently is implementation detail. We do this because we haven't implemented GC in layers other than data stores. You should mention this where you filter out other nodes and not here.


So, this function finds all new references or outbound routes in the current graph that haven't been explicitly notified to GC. The principle is that every new reference or outbound route must be notified to GC via the addedOutboundReference method. It it hasn't, its a bug and we want to identify these scenarios.